test/lib/y2firewall/clients/auto_test.rb

```diff
@@ -229,6 +228,82 @@
     end
   end
 
+  describe "#change" do
+    let(:result) { :ok }
+    let(:already_read) { false }
+    let(:exported_profile) { double("Exported Profile") }
+    let(:main_dialog) { instance_double("Dialog::Main", run: result) }
+
+    before do
+      allow(subject).to receive(:import)
+      allow(subject).to receive(:export).and_return(exported_profile)
+      allow(Y2Firewall::Dialogs::Main).to receive(:new).and_return(main_dialog)
+      allow(firewalld).to receive(:read?).and_return(already_read)
+    end
+
+    context "when the configuration is accepted" do
+      [:back, :abort, :cancel].each do |dialog_result|
+        let(:result) { dialog_result }
+
+        it "does not set autoyast config" do
+          expect(subject.class).to_not receive(:ay_config=)
+
+          subject.change
+        end
+      end
+    end
+
+    context "when the configuration is rejected" do
+      [:next, :finish, :ok, :accept].each do |dialog_result|
+        let(:result) { dialog_result }
+
+        it "sets autoyast config" do
+          expect(subject.class).to receive(:ay_config=).with(true)
+
+          subject.change
+        end
+      end
+    end
+
+    context "when firewalld is not installed yet" do
+      let(:installed) { false }
+
+      it "does not read its configuration" do
+        expect(firewalld).to_not receive(:read)
+
+        subject.change
+      end
+    end
+
+    context "when firewalld is installed" do
+      context "and its configuration was already read" do
+        let(:already_read) { true }
+
+        it "does not read it again" do
+          expect(firewalld).to_not receive(:read)
+
+          subject.change
+        end
+      end
+
+      context "but its configuation is not read" do
+        it "reads the minimal configuration" do
+          expect(firewalld).to receive(:read).with(minimal: true)
+
+          subject.change
+        end
+
+        it "keeps the previous configuration" do
```

Comment:
The scenario that we want to cover is when a configuration has been imported previously and then we modify it. We want to ensure that the configuration is not lost.

So would test it importing a minimal profile. For example:

```xml
<zones config:type="list">
  <zone>
    <name>example</name>
    <services config:type="list">
      <service>http</service>
      <service>https</service>
      <service>ssh</service>
    </services>
  </zone>
</zones>  
```

Reply:
Ok, let me improve/fix it :)

Thank you!

Reply:
Although from other POV it is more an integration test than a unit test, so if it is too complicate just let it as it is.

Reply:
Ok, take a look to https://github.com/yast/yast-firewall/pull/119/commits/19a1375f672a4be31d936689c83f0c87b0a6f4ae Does it look good for you? If so, I'll squash it in the previous commit. If not, I'll discard it and 

> just let it as it is.

Thanks!